### PR TITLE
Update Release Workflow

### PR DESCRIPTION
**Pull request by jdn5126**
**What type of PR is this?**
enhancement

**Which issue does this PR fix**:
N/A

**What does this PR do / Why do we need it**:
This PR updates the release workflow to run only for published releases. The workflow will check out the release tag and run `make` commands from that workspace.

Changes to the release scripts are as follows:
1. GitHub CLI and helm version is updated
2. `make upload-resources-to-github` is removed. We do not need to upload release artifacts since chart is available on `eks-charts` and manifest is available and updated in release branch.
3. `make generate-cni-yaml` is updated to no longer generate release artifacts.
4. `make config-folder-sync` is updated to only generate PR for master branch (not release branch).

**If an issue # is not available please add repro steps and logs from IPAMD/CNI showing the issue**:
N/A

**Testing done on this change**:
N/A

**Automation added to e2e**:
N/A

**Will this PR introduce any new dependencies?**:
No

**Will this break upgrades or downgrades. Has updating a running cluster been tested?**:
N/A

**Does this change require updates to the CNI daemonset config files to work?**:
No

**Does this PR introduce any user-facing change?**:
No

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: scripts/sync-to-config-folder.sh [hunk 1]
```diff
@@ -180,4 +179,4 @@ git push -u origin "${FORK_RELEASE_BRANCH}":$RELEASE_BRANCH
 gh pr create --title "🥳 ${BINARY_BASE} ${VERSION} Automated manifest sync! 🥑" \
     --body "${PR_BODY}" --repo ${REPO} --base ${RELEASE_BRANCH}
 
-echo "✅ Manifest folder PR created for $RELEASE_BRANCH"
+echo "✅ Manifest folder PR created for{RELEASE_BRANCH}"
```
Comment (orsenthil) [inline scripts/sync-to-config-folder.sh:182]: Should there be a `$` with {RELEASE_BRANCH}` ?

File: scripts/upload-resources-to-github.sh [hunk 2]
```diff
@@ -6,7 +6,6 @@ set -euo pipefail
 SCRIPTPATH="$( cd "$(dirname "$0")" ; pwd -P )"
 VERSION=$(make -s -f $SCRIPTPATH/../Makefile version)
 BUILD_DIR=$SCRIPTPATH/../build/cni-rel-yamls/$VERSION
-BINARY_DIR=$SCRIPTPATH/../build/bin
```
Comment (orsenthil) [inline scripts/upload-resources-to-github.sh:9]: Why this is not required now?

File: scripts/sync-to-config-folder.sh (hunk 1 above)
Comment (jdn5126) [reply scripts/sync-to-config-folder.sh:182]: Ah yes this was a typo, thank you for catching!

File: scripts/upload-resources-to-github.sh (hunk 2 above)
Comment (jdn5126) [reply scripts/upload-resources-to-github.sh:9]: I grepped and this was not used anywhere. I think we were at one time considering publishing binary artifacts (at this time we only publish compressed helm charts), but never ended up doing so.

File: charts/aws-vpc-cni/templates/clusterrole.yaml [hunk 3]
```diff
@@ -28,7 +28,7 @@ rules:
   - apiGroups: [""]
     resources:
       - nodes
-    verbs: ["list", "watch", "get"]
+    verbs: ["list", "watch", "get", "update"]
```
Comment (jayanthvn) [inline charts/aws-vpc-cni/templates/clusterrole.yaml:31]: I assume this will be removed for 1.13.0?

**Review by jayanthvn — APPROVED** [review]
lgtm

**Pull request merged**